src/WidgetBase.ts

```diff
@@ -385,6 +391,13 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			const { children, properties = {} } = dNode;
 			const { key } = properties;
 
+			const propagatedPropertyNames = this.getDecorator('propagateProperty');
```

Comment:
I talked with @matt-gadd a bit about this this morning. There doesn't seem to be a clever way to actually use `afterRender` here. We can't add it from the `propagateProperty` decorator function because then you can't add the decorator programatically. Also, running `propagateProperty` 5 times would add 5 `afterRender` hooks, each looping over the resulting node, which is less efficient than looping over once.

This may need a good think 😄